### PR TITLE
[OPPRO-330] Support velox/sparksql datetime functions in Gluten.

### PR DESCRIPTION
**Pull request by lviiii**
## What changes were proposed in this pull request?
Enable velox/sparksql Datetime functions, including day(), dayofweek(), dayofmonth(), dayofyear(), weekday(), month(), quarter(), year(), yearofweek(), and extract(QUARTER from xx)...
More details refer to https://github.com/oap-project/velox/pull/81

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/expression/UnaryOperatorTransformer.scala [hunk 1]
```diff
@@ -954,8 +940,8 @@ object UnaryOperatorTransformer {
       new ToRadiansTransformer(child, t)
     case r: Rint =>
       new RintTransformer(child, r)
-    case q: Quarter =>
-      new QuarterTransformer(child, q)
+//    case q: Quarter =>
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/expression/UnaryOperatorTransformer.scala:943]: please remove

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala [hunk 2]
```diff
@@ -60,13 +60,75 @@ class VeloxStringFunctionsSuite extends WholeStageTransformerSuite {
...
   }
 
-  ignore("day_of_month") {
+  test("extract") {
+    runQueryAndCompare(s"select l_orderkey, l_shipdate, " +
+      s"extract(doy FROM DATE'2019-08-12') " +
+      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+  }
+
+  ignore("second") {
```
Comment (rui-mo) [inline backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala:69]: Could you add a comment here to explain why this test is ignored for now?

**Review by rui-mo — APPROVED** [review]

File: gluten-core/src/main/scala/io/glutenproject/expression/DateTimeExpressionsTransformer.scala [hunk 3]
```diff
@@ -108,35 +111,40 @@ class DayOfMonthTransformer(child: Expression) extends DayOfMonth(child) with
...
+
+  override def getDataType: DataType = dataType
+
+  override def getChild: Expression = child
+
+}
+
+class WeekDayTransformer(child: Expression) extends WeekDay(child) with
+  ExtractDateTransformer {
+  override def getField: String = "DAY_OF_WEEK"
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/expression/DateTimeExpressionsTransformer.scala:138]: The return values of these two date function are different, is it ok to use the same field name?

Comment (lviiii) [reply gluten-core/src/main/scala/io/glutenproject/expression/DateTimeExpressionsTransformer.scala]: Updated.

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala (hunk 2 above)
Comment (lviiii) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala]: Done

**Pull request merged**